engines/native/system/process_posix.go

```diff
@@ -107,7 +106,11 @@ func killProcesses(root *process.Process) error {
 		}()
 	}
 
-	return nil
+	// if there was an error somewhere we return it
+	if err != nil {
+		return err
+	}
```

Comment:
I prefer to return `err` from parent if there is one, otherwise I return `err2` originating from killing a child process... If there is an `err2` at all..